_docs/guides/endpoints.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Install Istio for Google Cloud Endpoints Services  
+overview: Guideline to manually integrate Google Cloud Endpoints services with Istio.
+
+order: 42
+layout: docs
+type: markdown
+---
+{% include home.html %}
+
+This document provides a guideline to manually integrate Istio with existing
+Google Cloud Endpoints services. 
+
+## Before you begin
+
+You need to follow the instruction to setup an Endpoints service on GKE [instructions](https://cloud.google.com/endpoints/docs/openapi/get-started-kubernetes-engine)
+After setup, you should be able to get an API key and store it in `ENDPOINTS_KEY` environment variable and the external IP address `EXTERNAL_IP`.
+You may test the service using the following command: 
+```
+curl --request POST       --header "content-type:application/json"       --data '{"message":"hello world"}'       "http://${EXTERNAL_IP}:80/echo?key=${ENDPOINTS_KEY}"
+```
+
+You need to install Istio with [instructions]({{home}}/docs/setup/kubernetes/quick-start.html#google-kubernetes-engine).
+
+## HTTP Endpoints service.
+
+1. Inject the service into Istio mesh using `--includeIPRanges` so that Egress is allowed to call external services directly. 
+Otherwise, ESP won't be able to access Google cloud service control. Follow the [instructions]({{home}}/docs/tasks/traffic-management/egress.html#calling-external-services-directly)
+
+2. After injection, the same testing command by calling ESP directly continue to work. 
+
+3. If you want to access the service through Ingress, setup the following:
+```
+cat <<EOF | istioctl create -f -
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: simple-ingress
+  annotations:
+    kubernetes.io/ingress.class: istio
+spec:
+  rules:
+  - http:
+      paths:
+      - path: /echo
+        backend:
+          serviceName: esp-echo
+          servicePort: 80
+EOF
+```
+
+4. Get the Ingress IP through [instructions]({{home}}/docs/tasks/traffic-management/ingress.html#verifying-ingress)
+You can verify accessing the Endpoints service through Ingress:
+```
+curl --request POST       --header "content-type:application/json"       --data '{"message":"hello world"}'       "http://${INGRESS_HOST}:80/echo?key=${ENDPOINTS_KEY}"i
```

Comment:
same bash comment: https://github.com/istio/istio.github.io/blame/master/_docs/setup/kubernetes/quick-start-gke-dm.md#L74-L76

and whitespace removal depending on what your after with the render.